frontend-project/src/models/users.ts

```diff
@@ -0,0 +1,46 @@
+import { fetchAll } from '@/services/users';
+import { Effect, EffectsCommandMap } from 'dva';
+import { Reducer, AnyAction } from 'redux';
+
+export interface User {
+  username: string;
+  email: string;
+  firstName: string;
+  lastName: string;
+  id: number;
+}
+
+export interface UsersState {
+  users: User[];
+}
+
+export interface UsersModelType {
+  namespace: string;
+  state: UsersState;
+  effects: {
+    fetchAll: Effect;
+  };
+  reducers: {
+    saveAll: Reducer<UsersState>;
+  };
+}
+
+const UsersModel = {
+  namespace: 'users',
+  state: [],
```

Comment:
Jeżeli mamy
```
const UsersModel: UsersModelType = {
```
to tutaj
```
state: {
  users: [],
}
```

Reply:
A i ta zmiana (razem z całym chowaniem listy użytkowników pod `users`) będzie wymagało zmian też na stronie komponentu wykorzystującego ten store.